### PR TITLE
#184581180 Update User Profile

### PR DESCRIPTION
**Pull request by ericndungutse**
#### What does this PR do?

This PR allow user to get and update his profile. It also allow other authenticated users to get a certain user's profile

#### Description of Task to be completed?

Have the following endpoints working

- `GET /user/profile`
- `GET /user/<user_id>`
- `PATCH /user/updateMe` 

#### How should this be manually tested?

First make sure you have docker installed. In your terminal;

1. In your **e-comm-team-amigos-bn** directory, open terminal and run `git fetch origin pull/28/ft-user-profile-184581180`
2. Checkout to **ft-user-profile-184581180** branch
3. Run `npm install`
4. Run `npm run dev`
5. In postman or documentation do the following
     - Create Account / Signup
     - Use the returned token for authorizarion
     - Make requests on above mentioned endpoints.

#### Any background context you want to provide?

- Have Node Js installed in your local machine
- Should have express server running
- Should have your `.env` file with neccessary variables as in `.env.example`
     
#### What are the relevant pivotal tracker/Trello stories?

Story ID: [184581180](https://www.pivotaltracker.com/story/show/184581180)

### CODE REVIEW

File: src/controllers/authentication.js [hunk 1]
```diff
@@ -28,14 +28,14 @@ const createUser = async (req, res) => {
...
+
     return res.status(201).json({
       message: 'Account created successfully',
       data: user,
       token,
     });
   } catch (error) {
-    return res.status(500).json(error);
+    return res.status(500).json(error.message);
   }
```
Comment (Habinezajanvier) [inline src/controllers/authentication.js]: Make sure that you are not sending error object, sometimes it's not formatted an meangless to the user

File: src/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    res.status(200).json({
+      status: 'success',
+      profile: profile[1],
+    });
+  } catch (error) {
+    res.status(200).json({
+      status: 'fail',
+      message: error.message,
+    });
```
Comment (Habinezajanvier) [inline src/controllers/user.js]: Don't send message object in errors, send {error: error.message} instead

File: src/database/models/user.js [hunk 3]
```diff
@@ -15,11 +16,21 @@ module.exports = (sequelize, DataTypes) => {
...
+      gender: DataTypes.ENUM('Male', 'Female'),
+      role: {
+        type: DataTypes.ENUM('admin', 'vendor', 'normal'),
+        defaultValue: 'normal',
+      },
+      status: {
+        type: DataTypes.ENUM('active', 'inactive'),
+        defaultValue: 'active',
+      },
+      verified: { type: DataTypes.BOOLEAN, defaultValue: 'false' },
```
Comment (Habinezajanvier) [inline src/database/models/user.js]: Adding fields on db, make sure you add separate migrations for them

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

**Review by wilbrord2 — APPROVED** [review]

File: src/database/migrations/20230316212347-updateUserRelation.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      queryInterface.removeColumn('Users', 'billingAddress'),
+      queryInterface.removeColumn('Users', 'preferredLanguage'),
+      queryInterface.removeColumn('Users', 'birthdate'),
+      queryInterface.removeColumn('Users', 'preferredCurrency'),
+      queryInterface.removeColumn('Users', 'gender'),
+    ]);
+  },
+};
+
+// firstName: DataTypes.STRING,
```
Comment (nshutijonathan) [inline src/database/migrations/20230316212347-updateUserRelation.js]: commented out codes should not be pushed

**Review by nshutijonathan — COMMENTED** [review]
check comments I left

File: src/controllers/user.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  try {
+    res.status(200).json({
+      status: 'success',
+      profile: req.user,
+    });
+  } catch (error) {
+    res.status(400).json({
+      status: 'success',
+      message: error.message,
+    });
```
Comment (Emile-SHUMBUSHO) [inline src/controllers/user.js]: how can we have status of success yet we are catching error, use message when display success result otherwise use error

**Review by Emile-SHUMBUSHO — CHANGES_REQUESTED** [review]

**Comment by ericndungutse** [comment]
Accidentally closed.

**Pull request closed without merging**